discovery_engine_core/web-api/src/routes.rs

```diff
@@ -63,6 +64,25 @@ fn user_path() -> impl Filter<Extract = (UserId,), Error = Rejection> + Clone {
         })
 }
 
+/// Extract a "count" from query params and check if within bounds, or reject with InvalidCountParam error.
+fn with_count_query_param(
+    default_documents_count: usize,
+) -> impl Filter<Extract = (PersonalizedDocumentsQuery,), Error = Rejection> + Copy {
+    warp::query().and_then(
+        move |params: Option<PersonalizedDocumentsQuery>| async move {
```

Comment:
i made the `count` optional in `PersonalizedDocumentsQuery` and simplified the route a bit: the unwrap-or-default moved from the router to the handler, as the default value doesn't have to be validated again since it is already validated at initialization. therefor we also don't have to introduce another struct for the checked query.